server/src/main/kotlin/core/pokerhands/PokerHand.kt

```diff
@@ -0,0 +1,19 @@
+package core.pokerhands
+
+import core.Card
+
+abstract class PokerHand(val rank: HandRank, val cards: List<Card>) {
+
+    init {
+        // checks if cards parameter is a list of five distinct cards
+        assert((cards.size == 5) and (cards.distinct().size == 5))
+        assert(isValidHand())
```

Comment:
This assertion is shit

Read comment in `handUtils.kt` in line 23

Reply:
This is necessary to avoid creating illegal hands (e.g. 4-card hands) and in my opinion there's nothing smelly about this line.